source/runtime.hpp

```diff
@@ -288,6 +295,7 @@ namespace reshade
 		bool _last_reload_successful = true;
 		bool _should_save_screenshot = false;
 		bool _is_in_between_presets_transition = false;
+		bool _effect_preprocessor_modified = false;
```

Comment:
There is no need for a global variable like this. This is only ever set to true in one place, so just saving the preset there directly avoid all this bookkeeping.

Reply:
There was a request to also save the preset when the "reload" all the effects button was clicked, or when the user use the shortcut key combination to reload the effects, or when switching to performance mode. That's why i added this global var (in order to use it at several places in the gui.cpp and the runtime.cpp code).

Reply:
But why would it be necessary to save the preset when the user reloads effects? The preset is already saved when any value is modified (and should simply be saved when a preprocessor definition is modified too). So saving it again would be superfluous.

Reply:
OK. Made the simplification, but i kept the _effect_preprocessor_modified  var in order to toggle the apply button...